OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -403,3 +405,89 @@ void ElementPropertiesDialog::updateProperties()
   // accept the dialog
   accept();
 }
+
+/*
+ * event filter for mParameterLineEdits and mInputLineEdit
+ * to detect the focus out event, and update
+ * the default start values from modeldesctiption.xml
+ */
+bool ElementPropertiesDialog::eventFilter(QObject *pObject, QEvent *pEvent)
+{
+  QLineEdit *pLineEdit = qobject_cast<QLineEdit*>(pObject);
+
+  if (pLineEdit && pEvent->type() == QEvent::FocusOut) {
+    if (pLineEdit->text().isEmpty()) {
+      // search the lineEdit index in parameters or inputs
+      int parameterIndex = mParameterLineEdits.indexOf(pLineEdit);
+      int inputIndex = mInputLineEdits.indexOf(pLineEdit);
+
+      if (parameterIndex != -1) {
+        QString pParameterLabelText = mParameterLabels.at(parameterIndex)->text();
+        updateStartValues(pParameterLabelText, pLineEdit);
+      }
+
+      if (inputIndex != -1) {
+        QString pInputLabelText = mInputLabels.at(inputIndex)->text();
+        updateStartValues(pInputLabelText, pLineEdit);
+      }
+    }
+  }
+  return QWidget::eventFilter(pObject, pEvent);
+}
+
+/*
+ * helper function to updateStartValues read from modeldescription.xml
+ */
+void ElementPropertiesDialog::updateStartValues(QString pName, QLineEdit * pLineEdit)
```

Comment:
The function name is misleading - or I just get it wrong. It basically restores the value from the model description file, right? It deletes the actual value and restores it from the hard coded model description file (or uses a default "0" for everything which is not an FMU).

Reply:
uses a default 0 for everything, which is not an FMU

Reply:
change the function name to `restoreDefaultStartValue`

Reply:
this is fixed with https://github.com/OpenModelica/OpenModelica/pull/6857/commits/b6c74656dc0ab44f208186502b09e4071766895e